node/db/src/rocks.rs

```diff
@@ -40,13 +42,19 @@ impl RocksDb {
         }
         if let Some(compaction_style) = &config.compaction_style {
             db_opts.set_compaction_style(compaction_style_from_str(compaction_style).unwrap());
+            db_opts.set_disable_auto_compactions(false);
+        } else {
+            db_opts.set_disable_auto_compactions(true);
```

Comment:
This appears to be unrelated to the `--halt-after-import` flag. Could you explain the RocksDB changes in the PR description?